src/librustc_typeck/check/upvar.rs

```diff
@@ -259,7 +263,7 @@ impl<'a, 'gcx, 'tcx> AdjustBorrowKind<'a, 'gcx, 'tcx> {
             })
     }
 
-    fn adjust_upvar_borrow_kind_for_consume(&self,
+    fn adjust_upvar_borrow_kind_for_consume(&mut self,
                                             cmt: mc::cmt<'tcx>,
                                             mode: euv::ConsumeMode)
     {
```

Comment:
> The code below is, like, super-dubious. Why is the guarantor supposed to be a deref of anything when we are moving out of a closure?


I'll have to re-read it, but I imagine because we are hard-coding the closure upvar kind to `Fn`.